### DISCUSSION TITLE
Unable to download (previously uploaded) attachments via iOS app, OSX app, Firefox addon, chrome addon, WebGUI

### DISCUSSION THREAD

**Discussion by tchap2** [dani-garcia/vaultwarden#1271] [General]
### Subject of the issue

After successfully uploading document to Identity or Card (.pdf, .jpg..etc), unable to view and/or download such atachments on iOS app, OSX app, Firefox addon, chrome addon, and WebGUI.

iOS app error message: "Unable to download file"
PC standalone apps: gear/hourglass keeps spinning.

The attachements ARE located on server in own folder, with files names hashed.

### Your environment

UNRAID

* Bitwarden_rs version: 2.16.1

* Install method: Docker image
* Clients used:  Standalone  OSX and iOS, add-on Firefox/Chrome, WebGUI.
* Reverse proxy and version:  Letsencrypt, latest
* Version of mysql/postgresql:  Mariadb, 10.4.17
* Other relevant information:

### Steps to reproduce

After successfully uploading document to Identity or Card (.pdf, .jpg..etc), unable to view and/or download such atachments on iOS app, OSX app, Firefox addon, chrome addon, and WebGUI.

iOS app error message: "Unable to download file"
PC standalone apps: gear/hourglass keeps spinning.

Config FILE
{
  "domain": "http://bitwarden.mydomain.com",
  "disable_icon_download": false,
  "signups_allowed": false,
  "signups_verify": false,
  "signups_verify_resend_time": 3600,
  "signups_verify_resend_limit": 6,
  "invitations_allowed": true,
  "password_iterations": 100000,
  "show_password_hint": false,
  "admin_token": "MASKED_FOR_PRIVACY",
  "invitation_org_name": "Bitwarden_RS",
  "ip_header": "X-Real-IP",
  "icon_cache_ttl": 2592000,
  "icon_cache_negttl": 259200,
  "icon_download_timeout": 10,
  "icon_blacklist_non_global_ips": true,
  "disable_2fa_remember": false,
  "authenticator_disable_time_drift": false,
  "require_device_email": false,
  "reload_templates": false,
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "disable_admin_token": false,
  "_enable_yubico": true,
  "_enable_duo": false,
  "_enable_smtp": true,
  "smtp_host": "MASKED_FOR_PRIVACY",
  "smtp_ssl": true,
  "smtp_explicit_tls": false,
  "smtp_port": 2525,
  "smtp_from": "MASKED_FOR_PRIVACY",
  "smtp_from_name": "Bitwarden_RS",
  "smtp_username": "MASKED_FOR_PRIVACY",
  "smtp_password": "MASKED_FOR_PRIVACY",
  "smtp_timeout": 15,
  "_enable_email_2fa": false,
  "email_token_size": 6,
  "email_expiration_time": 600,
  "email_attempts_limit": 3
}

### Expected behaviour

Download/view attachments

**Comment by BlackDex** [comment]
You need to configure the correct `DOMAIN` value for you environment.
That is probably not configured correctly, which results in a bad download.
It needs to be the exact same is you have in your browser, or clients, including port number, https:// etc...

https://github.com/dani-garcia/bitwarden_rs/blob/175f2aeace6a6099cb3ea47d2de9968e764b5f43/.env.template#L194
